health_check/contrib/celery/backends.py

```diff
@@ -1,29 +1,69 @@
+from celery.app import default_app as app
 from django.conf import settings
 
 from health_check.backends import BaseHealthCheckBackend
-from health_check.exceptions import (
-    ServiceReturnedUnexpectedResult, ServiceUnavailable
-)
-
-from .tasks import add
+from health_check.exceptions import ServiceUnavailable
 
 
 class CeleryHealthCheck(BaseHealthCheckBackend):
-    def check_status(self):
-        timeout = getattr(settings, 'HEALTHCHECK_CELERY_TIMEOUT', 3)
+    CORRECT_RESPONSE = {"ok": "pong"}
+
+    def _check_active_queues(self):
+        defined_queues = app.conf.CELERY_QUEUES
+
+        if not defined_queues:
+            return
+
+        active_queues = []
+
+        for queues in list(
+            app.control.inspect(self.active_workers).active_queues().values()
+        ):
+            active_queues.extend([queue.get("name") for queue in queues])
+
+        for defined_queue in defined_queues:
+            if defined_queue.name not in active_queues:
+                self.add_error(
+                    ServiceUnavailable(
+                        f"No worker for Celery task queue {defined_queue.name}"
+                    ),
+                )
 
+    def _check_ping_result(self):
+        self.active_workers = []
+
+        for worker, response in self.ping_result[0].items():
+            if response != self.CORRECT_RESPONSE:
+                self.add_error(
+                    ServiceUnavailable(
+                        f"Celery worker {worker} response was incorrect"
+                    ),
+                )
+                continue
+            self.active_workers.append(worker)
+
+        if not self.errors:
+            self._check_active_queues()
+
+    def check_status(self):
+        timeout = getattr(settings, "HEALTHCHECK_CELERY_PING_TIMEOUT", 1)
         try:
-            result = add.apply_async(
-                args=[4, 4],
-                expires=timeout,
-                queue=self.queue
-            )
-            result.get(timeout=timeout)
-            if result.result != 8:
-                self.add_error(ServiceReturnedUnexpectedResult("Celery returned wrong result"))
+            self.ping_result = app.control.ping(timeout=timeout)
```

Comment:
To avoid setting instance attributes outside of `__init__` method, let's modify `_check_ping_result()`, so it will accept `ping_result` argument